### DISCUSSION TITLE
Lock face tags for an image

### DISCUSSION THREAD

**Discussion by oledid** [immich-app/immich#7785] [General]
To get face tagging up and running, I had to test a couple settings variants,  different input values for the face recognition models etc. Every time I did that, I needed to rescan (and remove all faces).

I want to suggest a feature: A way to say that "for this image, the face tagging is completed manually. Do not touch this image in the face recognition jobs, and keep the faces the way they are. Never clear them or change them."

**Comment by aisbergde** [comment]
But I am afraid, faces locked this way will not be compatible with other faces. My first intent was similar: why not keep the information about already tagged faces? But finally I see more issues than benefits:
- should a new tagging process add new faces, leading to duplicates, or should it exlude the marked assets from processing?
- if marked assets are excluded, how to learn from them?
- How to handle the parallel existence of tags, generated by different tagging processes, including manually tagged?

That's why my idea was, to allow not only one set of tagged faces, but to store also the information about the used tagging process. Then a user could switch between different tagging result sets, could compare, and so on, not loosing the valuable data. I have to search the issue number and will add it later here.

**Reply by aisbergde** [reply]
it was #6044

**Comment by oledid** [comment]
My thoughts:
Keep manually tagged faces, do not learn from them, make it possible to say that "this face is the same face" as new ai generated face groups.
